assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+	 * to multiple contexts).
+	 *
+	 * Does not error if any of the requested contexts do not exist, but will console.warn
+	 * a message about non-existant contexts.
+	 *
+	 * Does not error if the area slug provided does not exist, but will console.warn a
+	 * message about non-existent areas.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}         areaSlug     Widget Area's slug.
+	 * @param  {(string|Array)} contextSlugs Widget Context's slug(s).
+	 * @return {Object}                      Redux-style action.
+	 */
+	assignWidgetArea( areaSlug, contextSlugs ) {
+		let contextSlugsAsArray;
+		if ( typeof contextSlugs === 'string' ) {
+			contextSlugsAsArray = [ contextSlugs ];
+		} else {
+			contextSlugsAsArray = [ ...contextSlugsAsArray ];
+		}
+
+		return {
+			payload: { areaSlug, contextSlugs: contextSlugsAsArray },
+			type: ASSIGN_WIDGET_AREA,
+		};
+	},
+
+	/**
+	 * Creates a widget area with a given name and settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}             areaSlug           Widget Area's slug.
+	 * @param  {Object}             settings           Widget Area's settings.
+	 * @param  {number}             settings.priority  Priority for this widget area. Default: 10.
+	 * @param  {string}             settings.title     Title for this widget area.
+	 * @param  {string}             settings.subtitle  Subtitle for this widget area.
+	 * @param  {(string|undefined)} settings.icon      URL to SVG icon for this widget area.
+	 * @param  {string}             settings.style     Widget area style (one of "boxes", "composite").
+	 * @return {Object}                                Redux-style action.
+	 */
+	registerWidgetArea( areaSlug, settings ) {
+		invariant( areaSlug, 'areaSlug is required.' );
+		invariant( settings, 'settings is required.' );
+		invariant( settings.title, 'settings.title is required.' );
+		invariant( settings.subtitle, 'settings.subtitle is required.' );
+		invariant( settings.style, 'settings.style is required.' );
+
+		const settingsToUse = {
+			priority: 10,
+			...settings,
+		};
+
+		return {
+			payload: { areaSlug, settings: settingsToUse },
+			type: REGISTER_WIDGET_AREA,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case ASSIGN_WIDGET_AREA: {
+			const { areaSlug, contextSlugs } = payload;
+
+			const { contexts } = state;
+			contextSlugs.forEach( ( contextSlug ) => {
+				if ( contexts[ contextSlug ] === undefined ) {
+					return;
+				}
+
+				if ( ! contexts[ contextSlug ].includes( areaSlug ) ) {
+					contexts[ contextSlug ].push( areaSlug );
+				}
+			} );
+
+			return {
+				...state,
+				contexts,
+			};
+		}
+
+		case REGISTER_WIDGET_AREA: {
+			const { areaSlug, settings } = payload;
+
+			if ( state.areas[ areaSlug ] !== undefined ) {
+				global.console.warn( `Could not register widget area with slug "${ areaSlug }". Widget Area "${ areaSlug }" is already registered.` );
```

Comment:
Nitpicking, but let's not capitalize the "Area" since it's just a regular noun. Should be "Widget area...".